framework/test/mocha/unit/modules/chain/submodules/process_transactions.js

```diff
@@ -41,16 +80,435 @@ describe('ProcessTransactions', () => {
+		stateStoreMock.account.restoreSnapshot.reset();
+		stateStoreMock.account.createSnapshot.reset();
+		stateStoreMock.transaction.add.reset();
+	});
+
+	describe('constructor()', () => {
+		it('should create instance of module', async () => {
+			expect(processTransactions).to.be.instanceOf(ProcessTransactions);
+		});
+
+		it('should assign parameters correctly', async () => {
+			const library = ProcessTransactions.__get__('library');
+
+			expect(library.storage).to.be.eql(paramScope.components.storage);
+		});
+
+		it('should invoke the callback', done => {
+			const cb = (error, object) => {
+				expect(error).to.be.null;
+				expect(object).to.be.instanceOf(ProcessTransactions);
+				done();
+			};
+			new ProcessTransactions(cb, paramScope);
+		});
+	});
+
+	describe('onBind()', () => {
+		it('should assign params correctly to library', async () => {
+			const library = ProcessTransactions.__get__('library');
+
+			expect(library.modules.blocks).to.be.eql(paramScope.modules.blocks);
+		});
+	});
+
+	describe('validateTransactions()', () => {
+		const validResponse = { status: TransactionStatus.OK, id: trs1.id };
+		const invalidResponse = { status: TransactionStatus.FAIL, id: trs2.id };
+
+		beforeEach(async () => {
+			trs1.validate = sinonSandbox.stub().returns(validResponse);
+			trs2.validate = sinonSandbox.stub().returns(invalidResponse);
+		});
+
+		it('should invoke validate() on each transaction', async () => {
+			processTransactions.validateTransactions([trs1, trs2]);
+
+			expect(trs1.validate).to.be.calledOnce;
+			expect(trs2.validate).to.be.calledOnce;
+		});
+
+		it('should update responses for exceptions for invalid responses', async () => {
+			const exceptionStub = sinonSandbox.stub();
+			ProcessTransactions.__set__(
+				'updateTransactionResponseForExceptionTransactions',
+				exceptionStub
+			);
+
+			processTransactions.validateTransactions([trs1, trs2]);
+
+			expect(exceptionStub).to.be.calledOnce;
+			expect(exceptionStub).to.be.calledWithExactly(
+				[invalidResponse],
+				[trs1, trs2]
+			);
+		});
+
+		it('should return transaction responses', async () => {
+			const result = processTransactions.validateTransactions([trs1, trs2]);
+
+			expect(result).to.be.eql({
+				transactionsResponses: [validResponse, invalidResponse],
+			});
+		});
+	});
+
+	describe('checkPersistedTransactions()', () => {
+		it('should resolve in empty response if called with empty array', async () => {
+			const result = await processTransactions.checkPersistedTransactions([]);
+
+			expect(result).to.be.eql({ transactionsResponses: [] });
+		});
+		it('should invoke entities.Transaction to check persistence of transactions', async () => {
+			storageMock.entities.Transaction.get.resolves([trs1, trs2]);
+
+			await processTransactions.checkPersistedTransactions([trs1, trs2]);
+
+			expect(storageMock.entities.Transaction.get).to.be.calledOnce;
+			expect(storageMock.entities.Transaction.get).to.be.calledWithExactly({
```

Comment:
It's trivial but you can combine these two assertions with `.and` matcher. But it's fine this way too but would be more readable if we combine them?

Reply:
I prefer to have more atomic expectations.